client/ayon_maya/plugins/publish/validate_no_face_sets.py

```diff
@@ -0,0 +1,51 @@
+import ayon_maya.api.action
+import maya.cmds as cmds
+from ayon_core.pipeline.publish import (
+    OptionalPyblishPluginMixin,
+    PublishValidationError,
+    RepairAction,
+    ValidateContentsOrder,
+)
+from ayon_maya.api import plugin
+
+class ValidateNoFaceSets(plugin.MayaInstancePlugin,
+                         OptionalPyblishPluginMixin):
+    """Ensure the meshes don't have face sets """
+
+    order = ValidateContentsOrder
+    families = ['model']
+    label = 'No Face Sets'
+    actions = [ayon_maya.api.action.SelectInvalidAction,
+               RepairAction]
+    optional = False
+
+    @staticmethod
+    def get_invalid(instance):
+        invalid = []
+        members = instance.data["setMembers"]
+        members = cmds.ls(members,
+                          dag=True,
+                          shapes=True,
+                          type=("mesh", "nurbsCurve"),
+                          noIntermediate=True,
+                          long=True)
+        if not instance.data.get("writeFaceSets"):
+            invalid.append(members)
```

Comment:
This validation is incorrect - it does not check at all whether the meshes have component shader assignments. This will now always invalidate if there are any meshes (or curves!) which to me seems wrong.

1. First, curves can have no face/component shading assignments, right? (or even any shader assignment in maya?) so should not need to be checked.
2. We should only find the meshes that have face shader assignments to begin with?

Otherwise there's no reason to make this a validator and instead force the `writeFaceSets` to True always.

Reply:
For reference, here's a similar validator for maya looks: https://github.com/ynput/ayon-maya/blob/06aa2445b4df511225b6795b283dd974040c0b5f/client/ayon_maya/plugins/publish/validate_look_single_shader.py#L1-L57

Basically disallowing shaders per face to be assigned. You can perform a similar check here and consider only the meshes with multiple shaders to have face assignments.